### ISSUE TITLE
Windows7 & NVim: Getting cmd.exe error for every plugin on :PlugStatus

### ISSUE THREAD

**Issue by tbttfox** [junegunn/vim-plug#583]
Explain the problem here ...

I'm running neovim on Windows 7, and running `:PlugStatus` prints
```
    The filename, directory name, or volume label syntax is incorrect.
    PlugClean required.
```
for every plugin managed by vim-plug.

I managed to track down what was happening, but I don't know the best way to handle this.
Apparently quoting a filepath in the `system()` command fails for my setup.

Running `nvim -u NORC`
This fails: `:echo system('cd /d "c:\Windows"')`
And this succeeds: `:echo system('cd /d c:\Windows')`

I have removed the extra quoting from `s:shellesc()` in my local copy and everything seems to work, but I can't really test on any other systems

------------------------------

```
NVIM v0.2.0-479-ga8a673c2
Build type: Release
Compilation: C:/msys64/mingw64/bin/gcc.exe -Wconversion -O2 -DNDEBUG -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -D__USE_MINGW_ANSI_STDIO -D_WIN32_WINNT=0x0600 -Wvla -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -IC:/projects/neovim/build/config -IC:/projects/neovim/src -IC:/projects/neovim/.deps/usr/include -IC:/projects/neovim/.deps/usr/include -IC:/projects/neovim/.deps/usr/include -IC:/msys64/mingw64/include -IC:/msys64/mingw64/include -IC:/projects/neovim/build/src/nvim/auto -IC:/projects/neovim/build/include
Compiled by appveyor@APPVYR-WIN

Optional features included (+) or not (-): -acl   +iconv    -jemalloc -tui      
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM\sysinit.vim"
  fall-back for $VIM: "C:/Program Files (x86)/nvim/share/nvim"
```

- Type:
    - [x] Bug
    - [ ] Enhancement
    - [ ] Feature Request
    - [ ] Question
- OS:
    - [ ] All/Other
    - [ ] Linux
    - [ ] OS X
    - [x] Windows
- Vim:
    - [ ] Terminal Vim
    - [ ] GVim
    - [x] Neovim

**Comment by junegunn** [comment]
Thanks for looking into it. Is the problem specific to Neovim? Have you had a chance to try it on vanilla Vim?

**Comment by tbttfox** [comment]
I think it's Neovim specific.
It works with and without quotes in vim 7.4. I don't have vim 8 installed on this box, but if I remember correctly they both work there as well.

**Comment by junegunn** [comment]
I see, I have no experience with Neovim on Windows.

@justinmk Hi, do you have any clue on this?

**Comment by justinmk** [comment]
This is a known issue, I haven't looked into yet, but it's related to https://github.com/neovim/neovim/pull/5280#issue-174549578 (the top comment) and https://github.com/JuliaLang/julia/issues/13776. 

We're actively working on getting the Windows build up to par, any help in narrowing down this problem and providing additional test cases is appreciated.

The `:echo system('cd /d "c:\Windows"')` test case is helpful, thanks @tbttfox.

**Comment by justinmk** [comment]
`s:shellesc` seems a bit suspicious because (1) the escape character for cmd.exe is `^`, not `\`, and (2) un-escaped nested quotes actually are kinda supported there anyways. Though `:echo system('cd /d "c:\Windows"')` in win32 Nvim anyways, so that's irrelevant.

For powershell the escape char is backtick.

**Comment by justinmk** [comment]
This might be fixed by https://github.com/neovim/neovim/pull/5957/commits/402ee7f3ac806e1a37576f697154faa91c697aa1. I'll merge that sometime today.

**Comment by justinmk** [comment]
Hmm, maybe not. FWIW, `cd /?` specifically mentions that quotes are unnecessary:

```
CHDIR command does not treat spaces as delimiters, so it is possible to
CD into a subdirectory name that contains a space without surrounding
the name with quotes.  For example:

    cd \winnt\profiles\username\programs\start menu

is the same as:

    cd "\winnt\profiles\username\programs\start menu"

```

**Comment by junegunn** [comment]
@justinmk See https://github.com/junegunn/vim-plug/pull/13. I don't know if it was the best way to handle the issue, but it worked somehow and I didn't bother to look deeper into it. Not a windows guy, any help/suggestion on Windows side is appreciated.

**Comment by christianrondeau** [comment]
I have created an issue on neovim, since there were none specifically related to this problem: https://github.com/neovim/neovim/issues/6329

@junegunn, I can "fix" the issue by removing quotes and escaping in `s:shellesc` when it's neovim, but I'm not sure if it's worth fixing when the issue is in neovim... on the other side, this would make `vim-plug` work with neovim on Windows in the meantime.

This being said, if `vim-plug` *only* escapes paths passed to `cd`, removing escaping altogether might be a valid option... though not in cygwin, so something like `if has('win') && !has('win32unix')` might be required.

In short: it's going to be messy, so it might be best to wait for neovim, but if this is not solved there, I could help with the alternative.
